prometheus/go_collector.go

```diff
@@ -16,16 +19,31 @@ func NewGoCollector() *goCollector {
 			Name: "process_goroutines",
 			Help: "Number of goroutines that currently exist.",
 		}),
+		gcDesc: NewDesc(
+			"process_gc_duration_seconds",
+			"A summary of the GC invocation durations.",
+			nil, nil),
 	}
 }
 
 // Describe returns all descriptions of the collector.
 func (c *goCollector) Describe(ch chan<- *Desc) {
 	ch <- c.goroutines.Desc()
+	ch <- c.gcDesc
 }
 
 // Collect returns the current state of all metrics of the collector.
 func (c *goCollector) Collect(ch chan<- Metric) {
 	c.goroutines.Set(float64(runtime.NumGoroutine()))
 	ch <- c.goroutines
+
+	var stats debug.GCStats
+	stats.PauseQuantiles = make([]time.Duration, 4)
+	debug.ReadGCStats(&stats)
+
+	quantiles := make(map[float64]float64)
+	for idx, pq := range stats.PauseQuantiles {
+		quantiles[float64(idx+1)/float64(len(stats.PauseQuantiles))] = pq.Seconds()
```

Comment:
The quantile calculation of the client_golang Summary is not part of the equation here. We are simply mirroring the quantiles the Go standard library gives us. I'd assume they calculate max and min in the same way the calculate the remaining quantiles (probably without decay, but if they apply a decay, it will reasonably be the same for min and max). I'd say a "0" and a "1" quantile for min and max makes most sense in this context.